lib/internal/event_target.js

```diff
@@ -0,0 +1,392 @@
+'use strict';
+
+const {
+  ArrayFrom,
+  Object,
+  Map,
+  SafeWeakMap,
+  Set,
+  SymbolFor,
+} = primordials;
+
+const {
+  codes: {
+    ERR_INVALID_ARG_TYPE,
+    ERR_INVALID_THIS,
+    ERR_EVENT_RECURSION,
+  }
+} = require('internal/errors');
+const kRejection = SymbolFor('nodejs.rejection');
+
+const perf_hooks = require('perf_hooks');
+const { customInspectSymbol } = require('internal/util');
+const { inspect } = require('util');
+
+const events = new SafeWeakMap();
+const targets = new SafeWeakMap();
```

Comment:
Since Node supports private class instance fields since 12.4, could that be used here instead of a WeakMap to simplify the code? 🤔

Reply:
See: https://github.com/nodejs/node/pull/33556#discussion_r430012404 ;-) ... the short answer is yes, absolutely.

Reply:
Resolving this as a duplicate of the other comment. Definitely making this change but benchmarking as I go.